schemas/rule.yml

```diff
@@ -0,0 +1,83 @@
+---
+- name: rule
+  title: Rule
+  group: 2
+  short: Fields to capture details about rules used by observers and agents to generate alerts or other events.
+  description: >
+   Rule fields are used to capture the specifics of any observer or agent rules that generate alerts or other events..
+   Examples of data sources that would populate the rule fields include: platforms, network or host IDS/IPS, network 
+   firewalls, web access firewalls, url filters, endpoint detection and response (EDR) systems, etc.  
+  
+  type: group
+  fields:
+
+    - name: id
+      level: extended
+      type: keyword
+      short: Rule ID
+      description: >
+        A rule ID that is unique within the scope of an agent, observer, or other entity using the rule for detection of this event
+
+      example: 101 
+
+    - name: uuid
+      level: extended
+      type: keyword
+      short: Rule UUID
+      description: >
+        A rule ID that is unique within the scope of a set or group of agents, observers, or other entities using the rule for detection of this event.
```

Comment:
The distinction is not clear enough vs `rule.id` IMO. @enotspe or @MikePaquette any ideas here?

Are there actually situations where one would populate both `rule.id` and `rule.uuid`? If you have both, is `rule.id` less relevant? I'm asking because perhaps we can simplify this and specifically say `rule.id` may be used to capture the rule UUID when available, and otherwise capture whatever rule ID makes sense from the device.

Reply:
I'm wondering if an observer event (e.g. a single PA firewall log) will have UUID for e.g. a Panorama rule?

Reply:
Yeah I'd like the two descriptions to be a little more differentiated. But at least it's clear that uuid is needed. (thanks for the links, @enotspe 🙂)

Let's leave it as is, and improve as needed later.